### PR TITLE
Support XDG_CONFIG_HOME env

### PR DESCRIPTION
**Pull request by zchee**
Support `XDG_CONFIG_HOME` environment variable.
And fix hardcoded filepath(`/`) separator, exported `EditFile`, `ConfigFile` and `QueryFile`.

I'm a macOS user, but many command line tools supported `XDG_CONFIG_HOME`. So I'm glad if borg also supports that environment.
Note that added backward compatibility that uses the`$HOME/.borg` if already exists.

---

This includes small personal opinion. Feel free to reject this pull request if you don't like it.

### CODE REVIEW

File: conf/conf.go [hunk 1]
```diff
@@ -36,24 +37,45 @@ var (
...
 func init() {
+	borgDir := borgDir()
+
+	EditFile = filepath.Join(borgDir, "edit")
+	ConfigFile = filepath.Join(borgDir, "config.yml")
+	QueryFile = filepath.Join(borgDir, "query")
+
+	os.Mkdir(borgDir, os.ModePerm)
+	os.Create(EditFile)
+	if _, err := os.Stat(ConfigFile); os.IsNotExist(err) {
```
Comment (crufter) [inline conf/conf.go:57]: Maybe replace this with a loop?

**Comment by crufter** [comment]
LGTM, man, thanks!

Gonna test it later this day and merge.

File: conf/conf.go (hunk 1 above)
Comment (zchee) [reply conf/conf.go:57]: Got it.
BTW, before code is no check `edit` file, so should I confirm to that?
like,

```go
	for _, f := range []string{EditFile, ConfigFile, QueryFile} {
		if _, err := os.Stat(f); os.IsNotExist(err) {
			os.Create(f)
		}
	}
```

vs

```go
	os.Create(EditFile)
	for _, f := range []string{ConfigFile, QueryFile} {
		if _, err := os.Stat(f); os.IsNotExist(err) {
			os.Create(f)
		}
	}
```

**Review by anas10 — APPROVED** [review]
LGTM too

**Comment by zchee** [comment]
@crufter Ah, I do not yet fix the loop section, it's OK?
BTW, Thanks merged :)

**Comment by crufter** [comment]
@zchee Ofc it's ok ;) Stuff can always be fixed later :P 

No worries, thanks for th help, man!

**Pull request merged**